pkg/telemetry/stats_telemetry.go

```diff
@@ -0,0 +1,41 @@
+package telemetry
+
+import "sync"
+
+// StatsTelemetrySender contains methods needed for sending stats metrics
+type StatsTelemetrySender interface {
+	Count(metric string, value float64, hostname string, tags []string)
+}
+
+// StatsTelemetryProvider handles stats telemetry and passes it on to a sender
+type StatsTelemetryProvider struct {
+	sender StatsTelemetrySender
+	m      sync.RWMutex
+}
+
+var (
+	statsProvider = &StatsTelemetryProvider{}
+)
+
+// RegisterStatsSender regsiters a sender to send the stats metrics
+func RegisterStatsSender(sender StatsTelemetrySender) {
+	statsProvider.m.Lock()
+	defer statsProvider.m.Unlock()
+	statsProvider.sender = sender
+}
+
+// GetStatsTelemetryProvider gets an instance of the current stats telemetry provider
+func GetStatsTelemetryProvider() *StatsTelemetryProvider {
+	return statsProvider
+}
+
+// Count reports a count metric to the sender
+func (s *StatsTelemetryProvider) Count(metric string, value float64, tags []string) {
+	s.m.RLock()
+	defer s.m.RUnlock()
+	if s.sender == nil {
+		return
+	}
+
+	s.sender.Count(metric, value, "", tags)
```

Comment:
Confidence check: AFAIR this empty hostname is replaced by a good one in the sender, but did you make sure of it?

Reply:
Looks like it's done in `sendMetricSample`. 
I kept it empty initially because that's how this is also used in the compliance agent telemetry (which I used as a rough guide).